src/functor/object.js

```diff
@@ -1,21 +1,19 @@
 import { Functor } from '../functor';
-import { append } from '../semigroup';
-import { foldr } from '../foldable';
 import stable from '../stable';
 
-const { getPrototypeOf } = Object;
+const { getPrototypeOf, keys } = Object;
 
 Functor.instance(Object, {
   map(fn, object) {
-    let properties = foldr(function(properties, entry) {
-      return append(properties, {
-        [entry.key]: {
-          enumerable: true,
-          get: stable(() => fn(entry.value, entry.key))
-        }
-      });
-    }, {}, object);
-    let prototype = getPrototypeOf(object);
-    return Object.create(prototype, properties);
+
+    let descriptors = keys(object).reduce((acc, key) => {
```

Comment:
Let's rename `acc` to `descriptors`.

Reply:
> Now that you've made the change, is it worth keeping the benchmark around?
Maybe as a way to show what options have been considered?